cmd/coder/shell.go

```diff
@@ -142,10 +145,26 @@ func runCommand(ctx context.Context, envName string, command string, args []stri
 	go func() {
 		stdin := process.Stdin()
 		defer stdin.Close()
-		io.Copy(stdin, os.Stdin)
+		_, err := io.Copy(stdin, os.Stdin)
+		if err != nil {
+			cancel()
+		}
 	}()
-	go io.Copy(os.Stdout, process.Stdout())
-	go io.Copy(os.Stderr, process.Stderr())
-
-	return process.Wait()
+	go func() {
+		_, err := io.Copy(os.Stdout, process.Stdout())
+		if err != nil {
+			cancel()
```

Comment:
nitpick: If one of these closes both do... It's fair to just defer cancel() at the top of both

Reply:
I don't think it's good to depend on that behavior though...